### PR TITLE
chore: add selection methods section to form docs

### PR DESCRIPTION
**Pull request by nkrantz**
[DSYS-434](https://twilio-productivity.atlassian.net/browse/DS-434)

### CODE REVIEW

File: packages/paste-website/src/pages/components/form/index.mdx [hunk 1]
```diff
@@ -416,6 +416,59 @@ Make required and optional fields distinguishable. Try to avoid optional input f
 
 Text is the clearest way to indicate whether a field is required or optional. However, the required symbol is relatively well understood. If you use the required symbol to indicate required fields, you'll see a "Required" title on the symbol. If you're building for other languages, use the `i18nLabel` prop to translate the "Required" title.
 
+#### Selection methods
+
+##### Single selection
+
+Component                                                                           | When to use
+----------------------------------------------------------------------------------- | -----------------------------------------------------------------------------------------------------------------------------------
+[Radio Button Group](/components/radio-button-group)                                | Use it when you have a list of up to 6 fixed items, and users need to select only one at a time.
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/form/index.mdx:425]:
```suggestion
[Radio Group](/components/radio-group) or [Radio Button Group](/components/radio-button-group)                                | Use it when you have a list of up to 6 fixed items, and users need to select only one at a time.
```

File: packages/paste-website/src/pages/components/form/index.mdx [hunk 2]
```diff
@@ -416,6 +416,59 @@ Make required and optional fields distinguishable. Try to avoid optional input f
...
+[Singleselect Combobox](/components/combobox)                                      | Use it when you have more than 6 options in a fixed list and users need to select only one at a time.
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/form/index.mdx:426]:
```suggestion
[Select](/components/select) or [Singleselect Combobox](/components/combobox)                                      | Use it when you have more than 6 options in a fixed list and users need to select only one at a time.
```

File: packages/paste-website/src/pages/components/form/index.mdx [hunk 3]
```diff
@@ -416,6 +416,59 @@ Make required and optional fields distinguishable. Try to avoid optional input f
...
+  <Dont
+    title="Don't"
+    preview={false}
+    body="Don’t use conditional filtering for essential filter criteria."
+  />
+</DoDont>
+<DoDont>
+  <Do title="Do" preview={false} body="Only one conditional filter should be expanded at a time." />
+</DoDont>
+
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/form/index.mdx:471]:
```suggestion
```

**Review by roxanagomez — COMMENTED** [review]
suggestion(non-blocking): At the end of the [docs](https://paste-docs-git-form-docs-selection-methods-twilio.vercel.app/components/form#related-paste-patterns-and-page-templates), under the "Related Paste Patterns and Page Templates" section, add a link to the Filter Pattern. (Maybe more relevant once the new filters are out)

**Review by roxanagomez — APPROVED** [review]

**Pull request merged**